src/plugins/unified_histogram/public/layout/layout.tsx

```diff
@@ -70,6 +80,22 @@ export interface UnifiedHistogramLayoutProps extends PropsWithChildren<unknown>
    * Append a custom element to the right of the hits count
    */
   appendHitsCounter?: ReactElement;
+  /**
+   * Disable automatic refetching based on props changes, and instead wait for a `refetch` message
+   */
+  disableAutoFetching?: boolean;
+  /**
+   * Disable triggers for the Lens embeddable
+   */
+  disableTriggers?: LensEmbeddableInput['disableTriggers'];
+  /**
+   * Disabled action IDs for the Lens embeddable
+   */
+  disabledActions?: LensEmbeddableInput['disabledActions'];
+  /**
+   * Input observable
+   */
+  input$?: UnifiedHistogramInput$;
```

Comment:
Had a look and changes that make much sense to me 👍 
So some loud thinking: now we offer 2 ways to trigger fetching, 1 by props and 1 by the input observable 

How about just 1? The input observable could rule it all by being a BehaviorSubject like this

```TypeScript
const input$ = new BehaviorSubject({
  type: 'fetch',
  payload: {
    query,
    filter,
    timeRange,
  }
});
this would replace the props, and it's responsible for triggering fetches and providing the details, what to fetch. I think it would simplify the code, at least in my understanding.

However, from consumer side, for a simple use case, that needs no refetch, providing props is simpler. so that's an argument against the "only input$" way

Reply:
I definitely think this is worth discussing more at least to see if we even need to support both. Devon and I discussed the Building Block APIs recently, and there were some good points raised and we have some ideas about how a standard Building Block API could look, but we'll need to discuss more as a team when I'm back from PTO.

For now I think it could be valuable to get this PR merged once the tests are passing since it simplifies the Discover side of things, and then do a PR with followup changes once we've made some decisions. WDYT?

Reply:
Just an FYI, the tests should now be fixed but I still have a couple tests to write for new functionality before publishing for review. Hoping to get to that tomorrow before PTO.

Reply:
yes, let's discuss and align after merging, for sure!